rdflib/plugins/sparql/operators.py

```diff
@@ -955,6 +956,17 @@ def datetime(e):
     return e.toPython()
 
 
+def date(e) -> py_datetime.date:
```

Comment:
Somewhat related to the code guideline discussion over at #1159 - are type annotations welcomed?

Reply:
Some Type annotations have already made their way into the codebase via PRs in the last few months.
If we were still trying to target Python 3.5, I'd say no we cannot have them. But since our minimum Python version on rdflib v5.1.0 will be Python 3.6, then there is no harm having Type Annotations in the code.

General rules would be:
* If your new code has type annotations, leave them in, no need to remove them
* but don't add type annotations to old code or code unrelated to your feature or fix.